code/numerical_analysis/integral/src/integral_rectangle.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+
+/*Part of Cosmos by OpenGenus Foundation*/
+
+double
+integral_rectangle(double a, double b, double step, double (*f)(double))
+{
+    double fragment = (b - a) / step;
+    double res = step * (f(a) + f(b)) / 6.0;
+    int i;
+    for (i = 1; i <= fragment; ++i)
+        res += 4.0 / 6.0 * step * f(a + step * (i - 0.5));
+
+    for (i = 1; i <= fragment - 1; ++i)
+        res += 2.0 / 6.0 * step * f(a + step * i);
+    return (res);
+}
+
+int
+main(void)
+{
+    double a, b, step, res;
+    printf("Enter interval: \n");
+    printf("a: \n");
+    scanf("%lf", &a);
+    printf("b: \n");
+    scanf("%lf", &b);
+    printf("Enter integration step: ");
+    scanf("%lf", &step);
+    res = integral_rectangle(a, b, step,
+                             ({
```

Comment:
I don't think you should indent this much, but that is my viewpoint.

Reply:
I want to clearly distinguish using function in another function. What do you think about it?

Reply:
No problem 👍